### PR TITLE
refactor(SpokePoolClient): Relocate validateFillForDeposit

### PR DESCRIPTION
**Pull request by pxrl**
This function compares a Deposit with a Fill and doesn't relate to the SpokePool or SpokePoolClient. Move it to FillUtils instead.

Motivation: This helps to slim down the SpokePoolClient and is one enabler for further functionalities to be relocated.

### CODE REVIEW

File: src/clients/SpokePoolClient.ts [hunk 1]
```diff
@@ -49,18 +50,6 @@ export type SpokePoolUpdate = {
   searchEndBlock: number;
 };
 
-const FILL_DEPOSIT_COMPARISON_KEYS = [
```
Comment (pxrl) [inline src/clients/SpokePoolClient.ts:52]: Note: This array is touched by the across-2.5 branch (`message` field is added). If this goes into master as-is then I will follow up with the corresponding change to align across-2.5.

File: src/clients/SpokePoolClient.ts [hunk 2]
```diff
@@ -223,9 +212,7 @@ export class SpokePoolClient {
     if (depositWithMatchingDepositId === undefined) {
       return undefined;
     }
-    return this.validateFillForDeposit(fillCopy, depositWithMatchingDepositId)
-      ? depositWithMatchingDepositId
-      : undefined;
+    return validateFillForDeposit(fillCopy, depositWithMatchingDepositId) ? depositWithMatchingDepositId : undefined;
```
Comment (james-a-morris) [inline src/clients/SpokePoolClient.ts:215]: Does the above conditional need to be present now that we have a validate fill function?

**Review by james-a-morris — COMMENTED** [review]
Left a comment

File: src/clients/SpokePoolClient.ts (hunk 2 above)
Comment (pxrl) [reply src/clients/SpokePoolClient.ts:215]: Hm..I think so? `validateFillForDeposit()` doesn't expect to receive `deposit = undefined`, so we can't blindly `depositWithMatchingDepositId` in.

Do you see an opportunity to simplify even further?

**Review by james-a-morris — COMMENTED** [review]
Replied to a comment

File: src/utils/FillUtils.ts [hunk 3]
```diff
@@ -265,3 +277,18 @@ export function getUnfilledDeposits(
     });
   }
 }
+
+// Ensure that each deposit element is included with the same value in the fill. This includes all elements defined
+// by the depositor as well as the realizedLpFeePct and the destinationToken, which are pulled from other clients.
+export function validateFillForDeposit(fill: Fill, deposit?: Deposit): boolean {
+  if (deposit === undefined) {
+    return false;
+  }
```
Comment (mrice32) [inline src/utils/FillUtils.ts:286]: nit: why do we allow deposit to be optional?

**Review by mrice32 — APPROVED** [review]
LGTM!

File: src/utils/FillUtils.ts (hunk 3 above)
Comment (pxrl) [reply src/utils/FillUtils.ts:286]: Was discussed here: https://github.com/across-protocol/relayer-v2/pull/602#discussion_r1175397986. It saves multiple callers from doing the validation.

File: src/utils/FillUtils.ts [hunk 4]
```diff
@@ -11,6 +11,18 @@ import {
...
+const FILL_DEPOSIT_COMPARISON_KEYS = [
+  "amount",
+  "originChainId",
+  "relayerFeePct",
+  "realizedLpFeePct",
+  "depositId",
+  "depositor",
+  "recipient",
+  "destinationChainId",
+  "destinationToken",
```
Comment (pxrl) [inline src/utils/FillUtils.ts:23]:
```suggestion
  "destinationToken",
  "message",
```

**Review by nicholaspai — APPROVED** [review]
LGTM, long time change incoming

**Review by james-a-morris — APPROVED** [review]
LGTM

**Pull request merged**